api/client/commands.go

```diff
@@ -2636,39 +2636,64 @@ type containerStats struct {
 
 func (s *containerStats) Collect(stream io.ReadCloser) {
 	defer stream.Close()
+	type d struct {
+		s   *stats.Stats
+		err error
+	}
 	var (
 		previousCpu    uint64
 		previousSystem uint64
 		start          = true
 		dec            = json.NewDecoder(stream)
+		updates        = make(chan *d, 1)
```

Comment:
for a struct of a pointer and an error interface, you don't need to make it a chan of pointer, chan of struct should be enough. So `make(chan d, 1)`.